independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/util/MediaTypeHelper.java

```diff
@@ -133,19 +126,31 @@ public static int compareWeight(MediaType one, MediaType two) {
         return COMPARATOR.compare(one, two);
     }
 
-    public static boolean sameWeight(MediaType one, MediaType two) {
-        return COMPARATOR.compare(one, two) == 0;
-    }
-
     public static void sortByWeight(List<MediaType> types) {
-        if (types == null || types.size() <= 1)
+        if (hasAtMostOneItem(types)) {
             return;
+        }
         types.sort(COMPARATOR);
     }
 
+    private static boolean hasAtMostOneItem(List<MediaType> types) {
+        return types == null || types.size() <= 1;
+    }
+
+    /**
+     * Finds the best match according to the weight of the media types
+     * The parameters needs to be sorted, so a copy of these is made if necessary
+     * in order to avoid altering the input
+     */
     public static MediaType getBestMatch(List<MediaType> desired, List<MediaType> provided) {
-        sortByWeight(desired);
-        sortByWeight(provided);
+        if (!hasAtMostOneItem(desired)) {
+            desired = new ArrayList<>(desired);
+            sortByWeight(desired);
+        }
+        if (!hasAtMostOneItem(provided)) {
+            provided = new ArrayList<>(provided);
+            sortByWeight(provided);
+        }
```

Comment:
This is never used in a hot path? Because I could see how the extra allocations could slow things down.

Reply:
It is not, because I removed the calls to `getBestMatch` from the hot path. The remaining calls to it are obscure features